tiledb/sm/query/ast/query_ast.h

```diff
@@ -473,6 +483,11 @@ class ASTNodeExpr : public ASTNode {
    */
   const QueryConditionCombinationOp& get_combination_op() const override;
 
+  /**
+   * Reverse the query condition using De Morgan's law.
+   */
+  void negate() override;
```

Comment:
IMO this should return a copy with the negation applied, similar to combine, rather than mutating. (cc @abigalekim @lums658 as well)